terraform/modules/ecs/ecs.tf

```diff
@@ -43,7 +43,7 @@ resource "aws_ecs_service" "web" {
   name            = "${var.ecs_service_web_name}"
   iam_role        = "${aws_iam_role.ecs_role.arn}"
   cluster         = "${aws_ecs_cluster.cluster.id}"
-  task_definition = "${aws_ecs_task_definition.web.arn}"
+  task_definition = "${replace("${aws_ecs_task_definition.web.arn}", "/:${aws_ecs_task_definition.web.revision}$$/", ":${var.ecs_service_web_task_revision}")}"
```

Comment:
Do we need that `$$` at the end of the regex, or is that a typo?

Reply:
`$$` escapes to `$`, i.e. ensuring the string ends with the revision. Revisions are numbers, and the ARN also includes numbers earlier on, so we want to make sure we're replacing the correct instance of the number.

Reply:
Ah, it's an escape character, of course!